integration/core/test_cli.py

```diff
@@ -1397,3 +1400,50 @@ def test_engine_restart_after_sigkill(bin):  # NOQA
 
     cleanup_process(em_client)
     cleanup_process(rm_client)
+
+def test_replica_removal_and_recreation(bin):  # NOQA
+    em_client = ProcessManagerClient(INSTANCE_MANAGER_ENGINE)
+    engine_process = create_engine_process(em_client)
+    grpc_controller_client = ControllerClient(
+        get_process_address(engine_process))
+    get_controller_version_detail(grpc_controller_client)
+
+    rm_client = ProcessManagerClient(INSTANCE_MANAGER_REPLICA)
+    replica_dir1 = tempfile.mkdtemp()
+    replica_dir2 = tempfile.mkdtemp()
+    replica_process1 = create_replica_process(rm_client, REPLICA_NAME,
+                                              replica_dir=replica_dir1)
+    grpc_replica_client1 = ReplicaClient(
+        get_process_address(replica_process1))
+    cleanup_replica(grpc_replica_client1)
+    replica_process2 = create_replica_process(rm_client, REPLICA_2_NAME,
+                                              replica_dir=replica_dir2)
+    grpc_replica_client2 = ReplicaClient(
+        get_process_address(replica_process2))
+    cleanup_replica(grpc_replica_client2)
+
+    open_replica(grpc_replica_client1)
+    open_replica(grpc_replica_client2)
+    r1_url = grpc_replica_client1.url
+    r2_url = grpc_replica_client2.url
+    v = grpc_controller_client.volume_start(replicas=[
+        r1_url, r2_url,
+    ])
+    assert v.replicaCount == 2
+
+    # Remove the first replica and recreate it immediately.
+    grpc_controller_client.replica_delete(r1_url)
+
+    for i in range(RETRY_COUNTS):
```

Comment:
Add wait interval for this retry loop. Then the test doesn't need to retry for 100 times.

Reply:
I added the sleep and changed it to use RETRY_COUNTS_SHORT.